en/material-system/effect-2.x-to-3.0.md

```diff
@@ -0,0 +1,254 @@
+
+> This article will detail the considerations for upgrading Cocos Creator 2.x materials to v3.0.
+
+## 1. Introduction to the base design of the material system
+
+### 1.1 The material system framework of Cocos Creator
+
+The material system consists of four core classes from top to bottom: **Material**, **Effect**, **Technique** and **Pass**, whose relationship can be understood by the following class diagram:
+
+![effect](material.png)
+
+#### Material
+
+The Material asset can be seen as an asset instance of the EffectAsset in the scene, and its own configurable parameters are effectAsset, technique, defines, states.
+
+#### Effect
+
+An Effect asset represents a material type and is the most important core asset in the material system. To implement a custom shading effect in the engine, it is necessary to write a custom Effect.
+
+#### Technique
+
+The solution to accomplish a final effect is referred to as a rendering technique , and a technique can be fused by one or more passes.
+
+#### Pass
+
+A Pass is a single GPU draw, typically including a vertex shader and a slice shader, and there are many optional configuration parameters for Pass in Creator.
+
+### 1.2 Material Instance Panel
+
+The Material instance panel is the most intuitive material editing window for all developers, and all actual material instances are configured through it.
+
+The Material instance panel in Cocos Creator 3.0 is as follows:
+
+![effect](material3.png)
+
+The Material instance panel of Cocos Creator 2.x looks like this:
+
+![effect](material10.jpg)
+
+Notice from the above two figures, the instance panel in v3.0 is quite complex compared to v2.4, partly because of the increased complexity of the material configuration, and partly because of the enhanced functionality of the panel.
+
+#### The configurable items of the material panel are divided into five main types
+
+1. Effect asset: the drop-down box will list all the Effect assets in the current project, and developers can select the Effect asset used by the current material. Other properties will be updated when the Effect is switched.
+2. Technique rendering technique selection: the drop-down box will list all the Technique in the Effect asset currently in use, and there may be multiple Technique in the Effect asset, each Technique is suitable for different situations. For example, the Technique with less effect but better performance is more suitable for mobile platform.
+3. Macro options defined in Effect: these macros control whether certain code can be enabled or not (corresponding to pass by pass).
+4. the list of properties defined in Effect (dynamically opened according to the macro definition) and the use of different input boxes to accommodate different types of properties. The editable properties of the editor are generally the mapping of a uniform in the shader, and from v3.0 onwards it is also possible to specify the mapping of a component (using the target parameter in Effect).
```

Comment:
The list of properties defined in the Effect, dynamically chose according to the macro definition. They are shown in different input types as defined type in the effect, e.g. number, color. The editable properties are generally the mapping of uniforms used by the shader, from v3.0, it's possible to specify the mapping of a property to a component in vector uniform by using the target parameter in Effect.